cardano-api/src/Cardano/Api/IPC.hs

```diff
@@ -550,11 +624,98 @@ submitTxToNodeLocal connctInfo tx = do
         atomically $ putTMVar resultVar result
         pure (Net.Tx.SendMsgDone ())
 
+
+data LocalTxMonitoringResult mode
+  = LocalTxMonitoringTxExists
+      TxId
+      SlotNo -- ^ Slot number at which the mempool snapshot was taken
+  | LocalTxMonitoringTxDoesNotExist
+      TxId
+      SlotNo -- ^ Slot number at which the mempool snapshot was taken
+  | LocalTxMonitoringNextTx
+      (Maybe (TxInMode mode))
+      SlotNo -- ^ Slot number at which the mempool snapshot was taken
+  | LocalTxMonitoringMempoolSizeAndCapacity
+      Consensus.MempoolSizeAndCapacity
+      SlotNo -- ^ Slot number at which the mempool snapshot was taken
+
+data LocalTxMonitoringQuery mode
+  -- | Query if a particular tx exists in the mempool. Note that, the absence
+  -- of a transaction does not imply anything about how the transaction was
+  -- processed: it may have been dropped, or inserted in a block.
+  = LocalTxMonitoringQueryTx (TxIdInMode mode)
+  -- | The mempool is modeled as an ordered list of transactions and thus, can
+  -- be traversed linearly. 'LocalTxMonitoringSendNextTx' requests the next transaction from the
+  -- current list. This must be a transaction that was not previously sent to
+  -- the client for this particular snapshot.
```

Comment:
Is there state associated connection that tracks where the current transaction in the list is?

Reply:
There are only 3 available things you can get:
```
data ServerStBusy (kind :: StBusyKind) txid tx slot m a where
  SendMsgReplyNextTx
    :: Maybe tx
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy NextTx txid tx slot m a

  SendMsgReplyHasTx
    :: Bool
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy HasTx txid tx slot m a

  SendMsgReplyGetSizes
    :: MempoolSizeAndCapacity
    -> ServerStAcquired txid tx slot m a
    -> ServerStBusy GetSizes txid tx slot m 
    ```